opendrive2lanelet2/convertor/opendrive2lanelet2convertor.py

```diff
@@ -91,17 +92,46 @@ def convert_vertice_to_node(self,node_id,vertice):
         local_x, local_y = x, y
         return Node(node_id,lat,lon, local_x, local_y)
 
-    def calc_distance_node_tresh(self,n1,n2):
-        if(abs(n1.lat - n2.lat) < 0.000000001 and abs(n1.lon - n2.lon) < 0.000000001):
-            return True
-        else:
-            return False
-    
-    def calc_distance_node_isclose(self,n1,n2):
-        if(math.isclose(n1.lat, n2.lat, rel_tol=1e-10) and math.isclose(n1.lon, n2.lon, rel_tol=1e-10)):
-            return True
-        else:
-            return False
+    # The key for the dictionary will be of the form: 'lat_lon',
+    # where lat and lon are integers (lat here = input latitude / threshold) ,
+    # and each have 12 number slots associated.
+    # This number is set for the current threshold of 0.00000001,
+    # where the maximum longitude value of -180 turns into: -18000000000,
+    # filling up all 12 slots. ex: for the point [179.992125274, -45.12893761],
+    # the key is: '017999212527_-04512893761'
+    def key_from_latlon_prec(self, point):
+        return f'{int(np.round(point[0],0)):012d}' + '_' + f'{int(np.round(point[1],0)):012d}'
+
+
+    # Uses a dictionary, self.all_nodes_dict, which contains key-value pairs for all existing points
+    # If there is an existing point within the threshold distance, it returns True and the index of that point
+    # If not, it adds the point to the dictionary and returns False
+    def check_duplicate_hash_precise(self, new, tolerace=0.00000001):
+        # Grab 5 relative points {-1.1, -0.5, 0, 0.5, 1} in both latitude and longitude
+        new = new / tolerace
+        thresh_range = [-1.1, -0.5, 0, 0.5, 1.1]
+        for lat_threshold in thresh_range:
+            for lon_threshold in thresh_range:
+                threshold = np.array([lat_threshold, lon_threshold])
+                test = new + np.round(threshold, 8)
+
+                # The float -> int conversion will fail if the input value is infinite: float('inf')
+                # If that happens, asssume that it isn't a duplicate and return that is isn't a duplicate
+                try:
+                    test_key = self.key_from_latlon_prec(test)
+                except OverflowError:
+                    print('Lat or lon over max value')
```

Comment:
Probably best to print the value so that we can track it down in the map

Reply:
Currently, this function doesn't have the information to print anything specific about the error point, as it only gets passed the lat and lon. 

I can certainly pass that in and print it, I just figured it would be better to wait and come up with a more well-thought out solution in CAR-3198.